static/js/tools/download/preview.tsx

```diff
@@ -0,0 +1,164 @@
+
+interface PreviewProps {
+  selectedOptions: DownloadOptions;
+  isDisabled: boolean;
+}
+
+export function Preview(props: PreviewProps): JSX.Element {
+  const [previewData, setPreviewData] = useState<string[][]>(null);
+  const [errorMessage, setErrorMessage] = useState<string>("");
+  const csvParser = usePapaParse();
+  const csvUrl = useRef("");
+
+  useEffect(() => {
+    if (props.isDisabled && _.isEmpty(errorMessage)) {
+      return;
+    }
+    csvUrl.current = getCsvUrl();
+    fetchPreviewData();
+  }, [props, errorMessage]);
+
+  // We only want to show preview once preview data has been fetched.
+  const showPreview = _.isEmpty(errorMessage) && !_.isEmpty(previewData);
+  // 1st row of previewData will be the header and the rest are the data rows.
+  const header = showPreview ? previewData[0] : [];
+  const dataRows = showPreview
+    ? previewData.slice(1).filter((row) => row.length > NUM_DEFAULT_COL)
+    : [];
+  const numCols =
+    NUM_DEFAULT_COL +
+    Object.keys(props.selectedOptions.selectedStatVars).length * NUM_COL_PER_SV;
+  // Add a row at the bottom of the table with "..." in each cell
+  const emptyRow = new Array(numCols).fill("");
+
+  let cardClassName = "preview-container";
+  if (!_.isEmpty(errorMessage)) {
+    cardClassName += " preview-error";
+  } else if (props.isDisabled) {
+    cardClassName += " preview-disabled";
+  }
+  return (
+    <Card id={SECTION_ID} className={cardClassName}>
+      {errorMessage && <div>{errorMessage}</div>}
+      {showPreview && (
+        <>
+          <table>
+            <thead>
+              <tr>
+                {header.map((heading, idx) => {
+                  return <th key={"heading" + idx}>{heading}</th>;
+                })}
+              </tr>
+            </thead>
+            <tbody>
+              {dataRows.map((row, rowIdx) => {
+                return (
+                  <tr key={"row" + rowIdx}>
+                    {row.map((cell, cellIdx) => {
+                      return <td key={`row${rowIdx}cell${cellIdx}`}>{cell}</td>;
+                    })}
+                  </tr>
+                );
+              })}
+              <tr>
+                {emptyRow.map((_, idx) => {
+                  return <td key={"empty" + idx}>...</td>;
+                })}
+              </tr>
+            </tbody>
+          </table>
+          <Button
+            className="download-button"
+            color={"primary"}
+            disabled={props.isDisabled}
+            onClick={() => window.open(csvUrl.current)}
+          >
+            Download
+          </Button>
+        </>
+      )}
+      <div className="screen">
+        <div id="spinner"></div>
+      </div>
+    </Card>
+  );
+
+  function getCsvUrl(): string {
+    const svParam = Object.keys(props.selectedOptions.selectedStatVars).join(
+      "&statVars="
+    );
+    const minDate = props.selectedOptions.dateRange
+      ? props.selectedOptions.minDate
+      : DATE_LATEST;
```

Comment:
setting min date to "latest" isn't very intuitive - it should be clearly documented.

Reply:
done